concrete-core-fixture/src/fixture/glwe_ciphertext_consuming_retrieval.rs

```diff
@@ -0,0 +1,384 @@
+    type PreExecutionContext = (GlweCiphertextView,);
+    type PostExecutionContext = (Vec<Precision::Raw>,);
+    type Criteria = (Variance,);
+    type Outcome = (Vec<Precision::Raw>, Vec<Precision::Raw>);
+
+    fn generate_parameters_iterator() -> Box<dyn Iterator<Item = Self::Parameters>> {
+        Box::new(
+            vec![
+                GlweCiphertextRetrievalParameters {
+                    glwe_dimension: GlweDimension(1),
+                    polynomial_size: PolynomialSize(512),
+                },
+                GlweCiphertextRetrievalParameters {
+                    glwe_dimension: GlweDimension(2),
+                    polynomial_size: PolynomialSize(1024),
+                },
+                GlweCiphertextRetrievalParameters {
+                    glwe_dimension: GlweDimension(2),
+                    polynomial_size: PolynomialSize(2048),
+                },
+            ]
+            .into_iter(),
+        )
+    }
+
+    fn generate_random_repetition_prototypes(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+    ) -> Self::RepetitionPrototypes {
+    }
+
+    fn generate_random_sample_prototypes(
+        parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+    ) -> Self::SamplePrototypes {
+        let Self::Parameters {
+            glwe_dimension,
+            polynomial_size,
+        } = parameters;
+        let num_elements = glwe_dimension.to_glwe_size().0 * polynomial_size.0;
+        let proto_ciphertext_view = maker.transform_raw_to_ciphertext(
+            Precision::Raw::uniform_vec(num_elements),
+            *polynomial_size,
+        );
+        (proto_ciphertext_view,)
+    }
+
+    fn prepare_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+    ) -> Self::PreExecutionContext {
+        let (proto_ciphertext_view,) = sample_proto;
+        (maker.synthesize_glwe_ciphertext_view(proto_ciphertext_view),)
+    }
+
+    fn execute_engine(
+        _parameters: &Self::Parameters,
+        engine: &mut Engine,
+        context: Self::PreExecutionContext,
+    ) -> Self::PostExecutionContext {
+        let (ciphertext_view,) = context;
+        let raw_ciphertext = unsafe {
+            engine
+                .consume_retrieve_glwe_ciphertext_unchecked(ciphertext_view)
+                .to_vec()
+        };
+        (raw_ciphertext,)
+    }
+
+    fn process_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+        context: Self::PostExecutionContext,
+    ) -> Self::Outcome {
+        let (proto_ciphertext_view,) = sample_proto;
+        let (raw_ciphertext,) = context;
```

Comment:
Same question here.

Reply:
result of a consuming retrieval so no actual entity needing unsythesizing